File: app/src/main/java/com/example/scrolling/di/modules/MyApplicationComponent.java
package com.example.scrolling.di.modules;

import com.example.scrolling.application.MyApplication;

import dagger.Component;
import dagger.android.AndroidInjectionModule;
import dagger.android.AndroidInjector;

@Component(modules = { AndroidInjectionModule.class, MyApplicationModule.class})
public interface MyApplicationComponent extends AndroidInjector<MyApplication> {
}

File: app/src/main/java/com/example/scrolling/di/modules/MyApplicationComponent.kt
package com.example.scrolling.di.modules

import com.example.scrolling.application.MyApplication
import dagger.Component
import dagger.android.AndroidInjectionModule
import dagger.android.AndroidInjector

@Component(modules = [AndroidInjectionModule::class, MyApplicationModule::class, ViewModelModule::class])
interface MyApplicationComponent : AndroidInjector<MyApplication?>